lib/controller.rb

```diff
@@ -59,6 +59,18 @@ class SlowFood < Sinatra::Base
     erb :menu
   end
 
+  enable :sessions
+
+  post '/menu' do
+    @dish = Dish.get(params[:dishid].to_i)
+    @dish.save
+    @cart = Cart.new
+    @cart = @dish
+    @cart.update
+    @cart.save
```

Comment:
You only need either `.update` or `.save` - I believe `.update` takes some parameters so it would look like `@cart.update(total: 123)` or something.